### PR TITLE
Remove forward declarations

### LINKED ISSUE #23192: Obsolete forward declarations validParams<Material>
## Bug Description
There is one forward declaration `validParams<Material>` in `Material.h` that looks obsolete and results in a linking error:
https://github.com/idaholab/moose/blob/f9a2a7b396f3d5313c5b58e36362963a8347d88f/framework/include/materials/Material.h#L33-L34

## Steps to Reproduce
I tried to compile my MOOSE application with gcc 11.2.0 on our cluster and got the linker error
```
undefined reference to InputParameters validParams<Material>()
```

## Impact
The error prevents from compiling my code (though I can easily remove the forward declarations and everything gets fine).

### PR DESCRIPTION
**Pull request by vovannikov**
Closes #23192

### CODE REVIEW

**Comment by loganharbour** [comment]
@vovannikov thanks for your first contribution!

File: framework/include/materials/Material.h [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
 //* This file is part of the MOOSE framework
```
Comment (loganharbour) [inline framework/include/materials/Material.h:2]:
```suggestion
//* This file is part of the MOOSE framework
```

Please remove this newline

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: framework/include/materials/Material.h (hunk 1 above)
Comment (vovannikov) [reply framework/include/materials/Material.h]: Oops, I added it accidentally.

**Review by loganharbour — APPROVED** [review]

**Review by loganharbour — CHANGES_REQUESTED** [review]
Whoops - one more thing. You need to reference the issue https://github.com/idaholab/moose/issues/23192 in at least one of your commit messages. This is one of our SQA requirements.

See https://civet.inl.gov/job/1306624/:

```

##########################################################################
ERROR: Your patch does not contain a valid ticket reference! (i.e. #1234)
Merge branch 'del_old_validParams' of https://github.com/vovannikov/moose into test
Remove line
Remove forward declarations
##########################################################################
```

**Comment by vovannikov** [comment]
> Whoops - one more thing. You need to reference the issue #23192 in at least one of your commit messages. This is one of our SQA requirements.

Sorry for that, this is my first contribution here :) Thank you for the hint!

**Comment by loganharbour** [comment]
> > Whoops - one more thing. You need to reference the issue #23192 in at least one of your commit messages. This is one of our SQA requirements.
> 
> Sorry for that, this is my first contribution here :) Thank you for the hint!

No worries! We're happy to support fresh contributors! Thanks again!

**Review by loganharbour — APPROVED** [review]

**Pull request merged**